smaug/operators/padding_op.h

```diff
@@ -0,0 +1,115 @@
+#ifndef _OPERATORS_PADDING_OP_H_
+#define _OPERATORS_PADDING_OP_H_
+
+#include "smaug/core/backend.h"
+#include "smaug/core/operator.h"
+#include "smaug/core/tensor.h"
+// #include "smaug/core/tensor_utils.h"
+#include "smaug/core/workspace.h"
+
+namespace smaug {
+
+/** \ingroup Operators
+ * \brief Pad a given tensor in different dimension.
+ *
+ * This has a software-based implementation.
+ *
+ * @tparam Backend The Backend that sets Alignment.
+ */
+template <typename Backend>
+class PaddingOp : public Operator {
+   public:
+    PaddingOp(const std::string& name,
+              Workspace* workspace)
+            : Operator(name, OpType::Repeat, workspace){
+        inputs.resize(kNumInputs, nullptr);
+        outputs.resize(kNumOutputs, nullptr);
+    }
+
+    PaddingOp(const std::string& name,
+              Workspace* workspace,
+              int val)
+            : Operator(name, OpType::Repeat, workspace), padder(val){
+        inputs.resize(kNumInputs, nullptr);
+        outputs.resize(kNumOutputs, nullptr);
+    }
+
+    /** Set the number of padders of the Tensor along each dimension. */
+    void setPadder(const int& val) {
+        padder = val;
+        // set output size?
+    }
+
+    auto getPadder() { return padder; }
```

Comment:
Two issues here:

1. Avoid `auto` return types; it makes the API unnecessarily hard to read and understand. The reader should not need to look for where `padder` is defined to know what this returns.
2. The name "padder" suggests that it is an object doing padding, when in fact it is just your padding size. So this API (and the class member) should be named "padding_size".

Reply:
Thank you for your advice.